junit-platform-gradle-plugin/src/test/groovy/org/junit/platform/gradle/plugin/JUnitPlatformPluginFunctionalSpec.groovy

```diff
@@ -0,0 +1,272 @@
+	}
+
+	def "can be used with 'java-library' plugin"() {
+		given:
+		javaFile()
+		succeedingTestFile()
+		javaLibraryPlugin()
+
+		when:
+		BuildResult result = GradleRunner.create()
+			.withProjectDir(testProjectDir.root)
+			.withPluginClasspath(pluginClasspath)
+			.withArguments('build')
+			.build()
+
+		then:
+		result.task(':junitPlatformTest').outcome == TaskOutcome.SUCCESS
+		result.task(':build').outcome == TaskOutcome.SUCCESS
+		result.output.contains('1 tests successful')
+	}
+
+	def "failing test fails build with 'java' plugin"() {
+		given:
+		javaFile()
+		failingTestFile()
+		javaPlugin()
+
+		when:
+		BuildResult result = GradleRunner.create()
+			.withProjectDir(testProjectDir.root)
+			.withPluginClasspath(pluginClasspath)
+			.withArguments('build')
+			.buildAndFail()
+
+		then:
+		result.task(':junitPlatformTest').outcome == TaskOutcome.FAILED
+		result.output.contains('1 tests failed')
+	}
+
+	def "failing test fails build with 'java-library' plugin"() {
+		given:
+		javaFile()
+		failingTestFile()
+		javaLibraryPlugin()
+
+		when:
+		BuildResult result = GradleRunner.create()
+			.withProjectDir(testProjectDir.root)
+			.withPluginClasspath(pluginClasspath)
+			.withArguments('build')
+			.buildAndFail()
+
+		then:
+		result.task(':junitPlatformTest').outcome == TaskOutcome.FAILED
+		result.output.contains('1 tests failed')
+	}
+
+	private static String splitClasspath(List<File> dependencies) {
+		return dependencies
+			.collect { it.absolutePath.replace('\\', '\\\\') } // escape backslashes in Windows paths
+			.collect { "'$it'" }
+			.join(', ')
+	}
+
+	private void javaPlugin() {
+		buildFile << """
+apply plugin: 'java'
+
+sourceCompatibility = JavaVersion.VERSION_1_8
+targetCompatibility = JavaVersion.VERSION_1_8
+
+dependencies {
+	testCompile files(${splitClasspath(testCompileClasspath)})
+	testRuntime files(${splitClasspath(testRuntimeClasspath)})
+	// Use local dependencies so that defaultDependencies are not used
+	junitPlatform files(${splitClasspath(testRuntimeClasspath)})
+}
+
+junitPlatform {
+	filters {
+		engines {
+			include 'junit-jupiter'
+		}
+	}
+}
+"""
+	}
+
+	private void javaLibraryPlugin() {
+		buildFile << """
+apply plugin: 'java-library'
+
+sourceCompatibility = JavaVersion.VERSION_1_8
+targetCompatibility = JavaVersion.VERSION_1_8
+
+dependencies {
+	testImplementation files(${splitClasspath(testCompileClasspath)})
+	testRuntimeOnly files(${splitClasspath(testRuntimeClasspath)})
+	// Use local dependencies so that defaultDependencies are not used
+	junitPlatform files(${splitClasspath(testRuntimeClasspath)})
+}
+
+junitPlatform {
+	filters {
+		engines {
+			include 'junit-jupiter'
+		}
+	}
+}
+"""
+	}
+
+	private void javaFile() {
+		Path javaFile = Paths.get(testProjectDir.root.toString(), 'src', 'main', 'java', 'org', 'junit', 'gradletest', 'Adder.java')
+		Files.createDirectories(javaFile.parent)
+		javaFile.withWriter {
+			it.write(
+				'''
```

Comment:
I think it's more consistent if the `'''` are on the previous line immediately following `write(`.